openapi-cli/src/main/java/io/ballerina/openapi/generators/service/ParametersGenerator.java

```diff
@@ -144,6 +145,10 @@ public List<Node> generateResourcesInputs(Map.Entry<PathItem.HttpMethod, Operati
         // public type PayloadType string|json|xml|byte[]|CustomRecord|CustomRecord[] ;
         if (operation.getValue().getRequestBody() != null) {
             RequestBody requestBody = operation.getValue().getRequestBody();
+            if (requestBody.get$ref() != null) {
+                String requestBodyName = extractReferenceType(requestBody.get$ref());
+                requestBody = this.components.getRequestBodies().get(requestBodyName.trim());
+            }
```

Comment:
Don't we need to check if the requestBody reference exist under `requestBodies`? (otherwise the `requestBody` will be null at L150 and can cause NPE?)

Reply:
The purpose of the request body definitions in the global `components.requestBodies` section is to avoid the same request body code occurrence in multiple operations.  
```openapi 
paths:
  /pets:
    post:
      summary: Add a new pet
      requestBody:
        $ref: '#/components/requestBodies/PetBody'
  /pets/{petId}
    put:
      summary: Update a pet
      parameters: [ ... ]
      requestBody:
        $ref: '#/components/requestBodies/PetBody'
components:
  requestBodies:
    PetBody:
      description: A JSON object containing pet information
      required: true
      content:
        application/json:
          schema:
            $ref: '#/components/schemas/Pet'
```
and it is highly unlike to happen reference inside internal reference this way.  (currently, tool doesn't support external references. Therefore NPE won't occur for the external scenario as well)
```openapi
components:
  requestBodies:
    PetBody1:
        $ref: '#/components/requestBodies/PetBody'
    PetBody:
      description: A JSON object containing pet information
      required: true
      content:
        application/json:
          schema:
            $ref: '#/components/schemas/Pet'
```
Your point is right we can have references in the body section there is no validation rule for it. 🙂 . if there is a reference it won't give NPE since their first reference gave `requestBody` object including fields null value and already it's `.getContent()` became null and the process does not proceed to map it. 
 Somehow we can handle the reference resolving. Please find this fix for it [52e9f35](https://github.com/ballerina-platform/openapi-tools/pull/1039/commits/52e9f351c18d7bb33442cd543fd8b32479648842)

Reply:
+1 for the new approach 🙂